pallets/crowdloan/src/tests.rs

```diff
@@ -0,0 +1,219 @@
+
+use crate::{mock::*, Error};
+
+// set_admin_account
+
+#[test]
+fn test_set_admin_account() {
+	let admin = ACCOUNT_00;
+	let new_admin = ACCOUNT_01;
+
+	ExtBuilder::default()
+		.with_admin_account(admin.clone())
+		.build()
+		.execute_with(|| {
+			assert_eq!(Crowdloan::admin_account(), admin);
+
+			// Change admin
+			assert_ok!(Crowdloan::set_admin_account(
+				Origin::signed(admin.clone()),
+				new_admin.clone()
+			));
+
+			// Test new admin is the one set
+			assert_eq!(Crowdloan::admin_account(), new_admin);
+		});
+}
+
+#[test]
+fn test_set_admin_account_with_sudo() {
+	let admin = ACCOUNT_00;
+	let new_admin = ACCOUNT_01;
+
+	ExtBuilder::default()
+		.with_admin_account(admin.clone())
+		.build()
+		.execute_with(|| {
+			assert_eq!(Crowdloan::admin_account(), admin);
+
+			// Change admin with sudo account
+			assert_ok!(Crowdloan::set_admin_account(Origin::root(), new_admin.clone()));
+
+			// Test new admin is the one set
+			assert_eq!(Crowdloan::admin_account(), new_admin);
+		});
+}
+
+#[test]
+fn test_no_custom_admin_set() {
+	let admin = ACCOUNT_00;
+	let new_admin = ACCOUNT_01;
+
+	ExtBuilder::default().build().execute_with(|| {
+		assert_eq!(Crowdloan::admin_account(), admin);
+
+		// Change admin
+		assert_ok!(Crowdloan::set_admin_account(
+			Origin::signed(admin.clone()),
+			new_admin.clone()
+		));
+
+		// Test new admin is the one set
+		assert_eq!(Crowdloan::admin_account(), new_admin);
+	});
+}
+
+#[test]
+fn test_set_admin_account_bad_origin_error() {
+	let admin = ACCOUNT_00;
+	let other_admin = ACCOUNT_01;
+
+	ExtBuilder::default()
+		.with_admin_account(admin.clone())
+		.build()
+		.execute_with(|| {
+			assert_noop!(
+				Crowdloan::set_admin_account(Origin::signed(other_admin), admin),
+				BadOrigin
+			);
+		});
+}
+
+// set_new_contribution
+
+#[test]
+fn test_set_new_contribution() {
+	let admin = ACCOUNT_00;
+	let contributor = ACCOUNT_01;
+	let contribution = BALANCE_01;
+
+	ExtBuilder::default()
+		.with_admin_account(admin.clone())
+		.build()
+		.execute_with(|| {
+			assert!(Crowdloan::contributions(&contributor).is_none());
+			assert_ok!(Crowdloan::set_new_contribution(
+				Origin::signed(admin.clone()),
+				contributor.clone(),
+				contribution
+			));
+			assert_eq!(Crowdloan::contributions(&contributor), Some(contribution));
+		});
+}
+
+#[test]
+fn test_override_contribution() {
+	let admin = ACCOUNT_00;
+	let contributor = ACCOUNT_01;
+	let contribution = BALANCE_01;
+	let new_contribution = BALANCE_02;
+
+	ExtBuilder::default()
+		.with_admin_account(admin.clone())
+		.with_contributions(vec![(contributor.clone(), contribution)])
+		.build()
+		.execute_with(|| {
+			assert_eq!(Crowdloan::contributions(&contributor), Some(contribution));
+			assert_ok!(Crowdloan::set_new_contribution(
+				Origin::signed(admin.clone()),
+				contributor.clone(),
+				new_contribution
+			));
+			assert_eq!(Crowdloan::contributions(&contributor), Some(new_contribution));
```

Comment:
W: would be cool to test here too that the event contains the old and new contribution